config/critters-webpack-plugin.js

```diff
@@ -0,0 +1,304 @@
+    const document = style.ownerDocument;
+
+    // basically `.textContent`
+    let sheet = style.childNodes.length>0 && style.childNodes.map( node => node.nodeValue ).join('\n');
+
+    // store a reference to the previous serialized stylesheet for reporting stats
+    const before = sheet;
+
+    // Skip empty stylesheets
+    if (!sheet) return done;
+
+    const ast = css.parse(sheet);
+
+    // Walk all CSS rules, transforming unused rules to comments (which get removed)
+    visit(ast, rule => {
+      if (rule.type==='rule') {
+        // Filter the selector list down to only those matche
+        rule.selectors = rule.selectors.filter(sel => {
+          // Strip pseudo-elements and pseudo-classes, since we only care that their associated elements exist.
+          // This means any selector for a pseudo-element or having a pseudo-class will be inlined if the rest of the selector matches.
+          sel = sel.replace(/::?(?:[a-z-]+)([.[#~&^:*]|\s|\n|$)/gi, '$1');
+          return document.querySelector(sel, document) != null;
+        });
+        // If there are no matched selectors, remove the rule:
+        if (rule.selectors.length===0) {
+          return false;
+        }
+      }
+
+      // If there are no remaining rules, remove the whole rule.
+      return !rule.rules || rule.rules.length!==0;
+    });
+
+    sheet = css.stringify(ast, { compress: this.options.compress!==false });
+
+    return done.then(() => {
+      // If all rules were removed, get rid of the style element entirely
+      if (sheet.trim().length===0) {
+        sheet.parentNode.removeChild(sheet);
+      }
+      else {
+        // replace the inline stylesheet with its critical'd counterpart
+        while (style.lastChild) {
+          style.removeChild(style.lastChild);
+        }
+        style.appendChild(document.createTextNode(sheet));
+      }
+
+      // output some stats
+      const name = style.$$name ? style.$$name.replace(/^\//, '') : 'inline CSS';
+      const percent = (before.length - sheet.length) / before.length * 100 | 0;
+      console.log('\u001b[32mCritters: inlined ' + prettyBytes(sheet.length) + ' (' + percent + '% of original ' + prettyBytes(before.length) + ') of ' + name + '.\u001b[39m');
+    });
+  }
+};
+
+
+/** Recursively walk all rules in a stylesheet.
+ *  The iterator can explicitly return `false` to remove the current node.
+ */
+function visit(node, fn) {
+  if (node.stylesheet) return visit(node.stylesheet, fn);
+
+  node.rules = node.rules.filter(rule => {
+    if (rule.rules) {
+      visit(rule, fn);
+    }
+    return fn(rule)!==false;
+  });
+}
+
+
+/** Enhance an htmlparser2-style DOM with basic manipulation methods. */
+function makeDomInteractive(document) {
+  defineProperties(document, DocumentExtensions);
+  // Find the first <html> element within the document
+  document.documentElement = document.childNodes.filter( child => String(child.tagName).toLowerCase()==='html' )[0];
```

Comment:
Have you linted this file with semistandard? The spaces look suspicious here

Reply:
I'll switch over - right now it is using my fork but needn't be.